File: 🟡 Medium/2491 - Divide Players Into Teams of Equal Skill/_2491DividePlayersIntoTeamsofEqualSkill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);

        int l = 0, r = skill.length - 1;
        int start = skill[l] + skill[r];
        long res = 0;

        while (l < r) {
            if (skill[l] + skill[r] != start) {
                return -1;
            }

            res += (skill[l] * skill[r]);
            l++;
            r--;
        }

        return res;
    }
}

File: 🟡 Medium/2491 - Divide Players Into Teams of Equal Skill/_2491DividePlayersIntoTeamsofEqualSkill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()

        l = 0
        r = len(skill) - 1

        start = skill[l] + skill[r]
        res = 0

        while l < r:
            if skill[l] + skill[r] != start:
                return -1

            res += skill[l] * skill[r]
            l += 1
            r -= 1

        return res
